mp/game/momentum/resource/ui/HudStickybombs.res

```diff
@@ -2,37 +2,51 @@
 {
     "CHudStickybombs"
     {
-        "ControlName"		"CHudStickybombs"
-        "fieldName"		"CHudStickybombs"
-        "visible"		"1"
-        "enabled"		"1"
-        "xpos"          "0"
-        "ypos"          "0"
-        "wide"			"20"
-        "tall"			"22"
-        "pin_to_sibling" "HudStickyCharge"
-        "pin_to_sibling_corner" "6" //bottom center
-        "pin_corner_to_sibling" "4" //top center
-        "fgcolor_override"		"0 0 0 0"
-        "bgcolor_override"		"0 0 0 0"
+        "ControlName" "CHudStickybombs"
+        "fieldName"   "CHudStickybombs"
+        "visible" "1"
+        "enabled" "1"
+        "xpos" "c-20"
+        "ypos" "c+107"
+        "wide" "40"
+        "tall" "10"
+        "BgColor" "Blank"
+        "BoxDimension" "8"
     }
-    "StickybombsLabel"
+    "FirstStickybox"
     {
-        "ControlName"	"Label"
-        "fieldName"		"StickybombsLabel"
-        "xpos"			"0"
-        "ypos"			"0"
-        "wide"			"f0"
-        "proportionalToParent" "1"
-        "autoResize"	"1"
-        "visible"		"1"
-        "enabled"		"1"
-        "tabPosition"	"0"
-        "textAlignment"	"center"
-        "dulltext"		"0"
-        "brighttext"	"0"
-        "font"			"HudNumbersSmallBold"
-        "fgcolor"       "243 243 243 255"
-        "auto_tall_tocontents" "1"
+        "ControlName" "FirstStickybox"
+        "fieldName"   "Stickybox"
+        "xpos" "0" // position set in code
+        "ypos" "0" // position set in code
+        "wide" "0" // see "BoxDimension" above
+        "tall" "0" // see "BoxDimension" above
+        "NoStickyColor" "BlackHO"
+        "PreArmColor"   "BlackHO"
+        "DisabledColor" "MomentumRed"
+    }
+    "SecondStickybox"
+    {
+        "ControlName" "SecondStickybox"
+        "fieldName"   "Stickybox"
+        "xpos" "0" // position set in code
+        "ypos" "0" // position set in code
+        "wide" "0" // see "BoxDimension" above
+        "tall" "0" // see "BoxDimension" above
+        "NoStickyColor" "BlackHO"
+        "PreArmColor"   "BlackHO"
+        "DisabledColor" "MomentumRed"
+    }
+    "ThirdStickybox"
+    {
+        "ControlName" "ThirdStickybox"
+        "fieldName"   "Stickybox"
+        "xpos" "0" // position set in code
+        "ypos" "0" // position set in code
+        "wide" "0" // see "BoxDimension" above
+        "tall" "0" // see "BoxDimension" above
+        "NoStickyColor" "BlackHO"
+        "PreArmColor"   "BlackHO"
+        "DisabledColor" "MomentumRed"
```

Comment:
One of the things I'm trying to push towards is not having duplicated logic or resources, think of the components inside our frontend website code.

We have to think about what exactly the role of StickyBox is. A StickyBox is a panel that is rendered as a square that gets its background color animated to match the state of the associated stickybomb from the launcher. This means that its state is going to be managed entirely by the hud_stickybombs class. Think of the speedometer and its speedometer labels. The HUD element manages all the labels, but the labels given the state handle all the rendering and animation. That's what we should be trying to go for here.

With this in mind, we can start to think about how best to define StickyBox. The HUD element is going to be constantly checking the status of the stickybombs and setting internal state of each of these StickyBoxes. But the catch is we want every single StickyBox to behave the exact same given the same internal state. I know there's swapping, but why not just swap the location of the panel instead of internal states?

The goal is to get rid of the idea of three distinct controls being defined and animated etc. They should all have the same panel name, have their settings handed to them by the hud panel, and all the res files shouldn't contain duplicates. Keep in mind you can always define colors in the scheme for use in specific panels, which are then used in animations as well, if need be. Another thing to figure out is running an animation on an individual control, as we'd prefer that to happen compared to three animations animating three different state variables in the HUD class. Think of the StickyBox as tied to a particular stickybomb.